appsec/src/helper/sampler.hpp

```diff
@@ -82,10 +87,38 @@ class sampler {
         return result;
     }
 
+    static double valid_sample_rate(double sampler_rate)
+    {
+        // NOLINTBEGIN(cppcoreguidelines-avoid-magic-numbers,readability-magic-numbers)
+        if (sampler_rate <= 0) {
```

Comment:
This is okay, I suppose you could also check that the status is valid in the listener.